readthedocs/projects/forms.py

```diff
@@ -634,6 +634,19 @@ def clean_domain(self):
             domain_string = parsed.netloc
         else:
             domain_string = parsed.path
+
+        if (
+            settings.PRODUCTION_DOMAIN and
+            settings.PRODUCTION_DOMAIN in domain_string
```

Comment:
I think we probably want a `endswith` in both of these, to reduce false positives.